js/reference.js

```diff
@@ -473,6 +494,43 @@ var ERMrest = (function(module) {
                         this._referenceColumns.push(compositeFKs[i]);
                     }
                 }
+
+                // if edit context remove filename, bytecount, md5, and sha256 from visible columns
+                if (module._isEntryContext(this._context) && assetColumns.length !== 0) {
+                    // given a colName will remove it from visible columns
+                    var removeCol = function(col) {
+                        // property does not exist
+                        if (col === null) {
+                            return;
```

Comment:
You can define an asset column, without defining any `filename`, `bytecolumn`, `md5`, or `sha256` columns for it. It will take care of that case. It think my comments are misleading. I will change them.